packages/editor/src/client/routes/polls/pollEditView.tsx

```diff
@@ -0,0 +1,256 @@
+import {ApolloError} from '@apollo/client'
+import React, {useEffect, useState} from 'react'
+import {useTranslation} from 'react-i18next'
+import {useNavigate, useParams} from 'react-router-dom'
+import {Col, DatePicker, FlexboxGrid, Form, Message, Panel, Row, Schema, toaster} from 'rsuite'
+
+import {
+  FullPoll,
+  PollAnswer,
+  PollAnswerWithVoteCount,
+  PollExternalVote,
+  PollExternalVoteSource,
+  usePollQuery,
+  useUpdatePollMutation
+} from '../../api'
+import {ModelTitle} from '../../atoms/modelTitle'
+import {PollAnswers} from '../../atoms/poll/pollAnswers'
+import {PollExternalVotes} from '../../atoms/poll/pollExternalVotes'
+
+export function PollEditView() {
+  const params = useParams()
+  const navigate = useNavigate()
+  const [poll, setPoll] = useState<FullPoll | undefined>(undefined)
+  const [close, setClose] = useState<boolean>(false)
+  const closePath = '/polls'
+
+  /**
+   * Handling toasts
+   */
+  const onErrorToast = (error: ApolloError) => {
+    toaster.push(
+      <Message type="error" showIcon closable duration={3000}>
+        {error.message}
+      </Message>
+    )
+  }
+  const onCompletedToast = () => {
+    toaster.push(
+      <Message type="success" showIcon closable duration={3000}>
+        {t('pollEditView.savedSuccessfully')}
+      </Message>
+    )
+  }
+
+  // get polls
+  const {data, loading: createLoading, refetch} = usePollQuery({
+    variables: {
+      pollId: params.id
+    },
+    onError: onErrorToast,
+    fetchPolicy: 'no-cache'
+  })
+
+  // updating poll
+  const [updatePoll, {loading: updateLoading, data: updateData}] = useUpdatePollMutation({
+    onError: onErrorToast,
+    onCompleted: onCompletedToast
+  })
+  const {t} = useTranslation()
+  const loading = createLoading || updateLoading
+
+  /**
+   * Update poll object after fetching from api
+   */
+  useEffect(() => {
```

Comment:
using `useState` instead of `useMemo` causes a double render here, can we switch to useMemo instead?